### ISSUE TITLE
how to publish a DAOD to a live deployment.

### ISSUE THREAD

**Issue by Joai2** [IBM-i2/Analyze#23]
Hello

I am trying to publish my DAOD to live deployment, I followed the instruction in the "example DAOD" but when I'm serching in "full text search" I get an error: "Server could not be found".

Do I have to export the project like the older versions?

**Comment by AamirShipton** [comment]
Hi Joai2,

I cannot answer you fully without knowing how your DAOD connector differs from our example. However, if you followed the instructions of copying the configuration directory from the development toolkit to the production toolkit, and deploying the production toolkit, the only thing left to do is restart your HTTP server. No need to restart the Liberty server. Just try the connector again.

**Comment by Joai2** [comment]
Hi Aamir

I restarted the HTTP and it is the same, I followed the same instructions. How does the topology lead to my DAOD project?

**Comment by Joai2** [comment]
Aamir,

I have the log of the error, it seems a problem with "ExampleDataAdapterSubset" the class is not found.

[messages.log.txt](https://github.com/IBM-i2/Analyze/files/58698/messages.log.txt)

**Comment by AamirShipton** [comment]
Hi Joai2,

Okay, that is definitely an issue that needs fixing before you can proceed. If the class is missing, you will first need to check that the JAR file of your connector project contains the ExampleDataAdapterSubset.class file (use 7-Zip, or similar, to inspect its contents). You should find this JAR file here:
C:\IBM\i2analyze\SDK\sdk-projects\master\build\toolkit\configuration\fragments\<your-project>\WEB-INF\lib\<your-project>.jar

If that JAR has the classes you expect in it, then you will need to check that the JAR was deployed to your project's app in your Liberty server. By default the path to this would be:
C:\IBM\i2analyze\deploy-dev\wlp\usr\servers\i2analyze\apps\<your-project>.war\WEB-INF\lib\<your-project>.jar

If this doesn't fix your problem, please confirm that you followed the instructions for moving the example to production. In particular, did you copy your development toolkit's configuration directory to the production toolkit?

**Comment by Joai2** [comment]
Hello Aamir

My project name is DAOD1. I checked the location of the "DAOD1.jar" and doesn't exist in the location, I found it in "C:\IBM\i2analyze\SDK\sdk-projects\DAOD1\build\libs", I will copy in the production server and test it. Is something wrong with my DAOD that the .jar doesn't exist in the location?

**Comment by AamirShipton** [comment]
Warning: I strongly recommend that you do not start changing the production system without first having your development environment working. That is extrememly risky as you can get yourself into a situation where the server is in a unusable state and you need to clear all of the data to fix the problem.

Yes, there is something wrong if your project isn't building the JAR file for the project to C:\IBM\i2analyze\SDK\sdk-projects\master\build\toolkit\configuration\fragments\DAOD1\WEB-INF\lib\DAOD1.jar

Have you altered master\build.gradle in any way?

**Comment by Joai2** [comment]
Yes Aamir, I changed it to add my DAOD1 project, but after the project  was added, I deleted all changes, so right now it is the same.
The changes made were only copy the script of "da-subset-filesystem-example" and add for "DAOD1".

**Comment by AamirShipton** [comment]
I will need some more detailed information about those changes as they almost certainly caused the problem. Please tell me the following:

1) What commands did you execute before you changed build.gradle?
2) Exactly what changes did you make to build.gradle?
3) What commands did you run with the modified build.gradle?
4) Why did you need to remove the changes to the build.gradle?
5) What commands did you run after you deleted your changes to build.gradle?

**Comment by Joai2** [comment]
Hello Aamir

1) I executed the command:
```
build -pr DAOD1 -t addDaodDataSource.
```
The command didn't execute because it sended an error of project not found, then I changed the settings.gradle and the build.gradle.

2) In the settings.gradle I added at the end of the last line:
```
, 'DAOD1'
```
and in the build.gradle I added:
```
project(':DAOD1') {
    dependencies {
        compile project(':da-example-common')
    }
}
```

3)After I added the changes in both files, I executed the same command:
```
build -pr DAOD1 -t addDaodDataSource.
```
It executed successfully but when I try to execute the command:
```
build -t deploy
```
I had an error with the build.gradle.

4) Because of the error, I had to remove the changes I did before.

5) After the changes were deleted, I executed the command
```
build -t deploy
```
And I didn't get any error.

**Comment by AamirShipton** [comment]
I can see nothing wrong with what you did until the error in point 3. Unfortunately, instead of fixing the error, you removed the changes you made to the Gradle scripts. Taking out your changes effectively removed your project from the list of projects that are built when you run 'build -t deploy'.
So you've removed the problem (hence you get no error now) but you will not be able to move to production without fixing this issue.

I suggest that you put back your changes to settings.gradle and build.gradle. You need DAOD1 to be included in both those files.
Next please send me a copy of the error that you receive when running 'deploy'. We can then try to help you fix the error and get back to a working state.

**Comment by Joai2** [comment]
Hello Aamir

I also made changes in the data schema, it is all different. And when I execute the deploy with DAOD added in settings.gradle and build.gradle I have this error:

[deploy.log.txt](https://github.com/IBM-i2/Analyze/files/77605/deploy.log.txt)

**Comment by Joai2** [comment]
I have not solved this issue, here are the log with -stacktrace, -debug and -info.
[buildDAOD1stacktrace.log.txt](https://github.com/IBM-i2/Analyze/files/95142/buildDAOD1stacktrace.log.txt)
[buildDAOD1info.log.txt](https://github.com/IBM-i2/Analyze/files/95141/buildDAOD1info.log.txt)
[buildDAOD1debug.log.txt](https://github.com/IBM-i2/Analyze/files/95140/buildDAOD1debug.log.txt)

**Comment by Joai2** [comment]
Issue solved.

The deployment in production doesn't work if the schema is different from the examples.
